tf_encrypted/config.py

```diff
@@ -82,306 +76,210 @@ def get_tf_config(self) -> Tuple[str, tf.ConfigProto]:
+        self._players = OrderedDict(
+            (name, Player(
                 name=name,
                 index=index,
                 device_name='/job:{job_name}/replica:0/task:{task_id}/cpu:0'.format(
                     job_name=job_name,
                     task_id=index
                 ),
                 host=host
-            )
-            for index, (name, host) in enumerate(hostmap)
-        }
+            ))
+            for index, (name, host) in enumerate(hostmap.items())
+        )
 
-    @classmethod
-    def from_dict(cls, params: Dict) -> Optional['RemoteConfig']:
+    @staticmethod
+    def load(filename):
         """
-        from_dict(params) -> RemoteConfig
+        Constructs a RemoteConfig object from a JSON hostmap file.
 
-        Produces a RemoteConfig class from a dictionary.
-
-        :param dict params: Key-value store of constructor arguments.
+        :param str filename: Name of file to load from.
         """
-        if not params.get('type', None) == 'remote':
-            return None
+        with open(filename, 'r') as f:
+            hostmap = json.load(f, object_pairs_hook=OrderedDict)
+        return RemoteConfig(hostmap)
 
-        return RemoteConfig(
-            hostmap=params['hostmap'],
-            job_name=params['job_name'],
-            master=params.get('master', None)
-        )
+    def save(self, filename):
+        """
+        Saves the configuration as a JSON hostmap file.
 
-    def to_dict(self) -> Dict:
-        hostmap = [(p.name, p.host) for p in sorted(self._players.values(), key=lambda p: p.index)]
-        params = {
-            'type': 'remote',
-            'job_name': self._job_name,
-            'hostmap': hostmap
-        }
+        :param str filename: Name of file to save to.
+        """
+        with open(filename, 'w') as f:
+            json.dump(self.hostmap, f)
 
-        if self._master is not None:
-            params['master'] = self._master
+    @property
+    def hostmap(self):
+        return OrderedDict(
+            (player.name, player.host)
+            for player in self._players.values()
+        )
 
-        return params
+    @property
+    def hosts(self):
+        return [
+            player.host
+            for player in self._players.values()
+        ]
 
     @property
-    def players(self) -> List[Player]:
+    def players(self):
         return list(self._players.values())
 
-    def get_player(self, name: str) -> Player:
+    def get_player(self, name):
         return self._players.get(name)
 
-    def get_players(self, names: Union[List[str], str]) -> List[Player]:
+    def get_players(self, names):
         if isinstance(names, str):
             names = [name.strip() for name in names.split(',')]
         assert isinstance(names, list)
         return [player for name, player in self._players.items() if name in names]
 
-    def server(self, name: str) -> tf.train.Server:
+    def server(self, name, start=True):
```

Comment:
Just curious if there's somewhere where we'd want to not start the server on creation?

Reply:
I'm not sure no. My reasoning was that the TF server takes this as a parameter, and `python -m tf_encrypted.player` makes an assumption that start is true; so if this were to ever change our launching of servers would be broken.